### ISSUE TITLE
Github action to build docker container occasionally cancelled

### ISSUE THREAD

**Issue by gassmoeller** [geodynamics/aspect#4969]
This was mentioned by @tjhei. Our github action that  builds the docker container for the official ASPECT dev version is occasionally cancelled (like here: https://github.com/geodynamics/aspect/actions/runs/3129691411/jobs/5079085394).
I investigated and noticed that this is on purpose. The github action will be automatically cancelled if another pull request is merged while the older is built. In this case it would be a bit of a waste to finish building the old container if we immediately start building another one (not to mention what would happen if two were built and pushed in parallel, I dont know how docker hub would react to that). So currently this action will sometimes be cancelled if two PRs are merged close after each another by this command in the dockerfile: https://github.com/geodynamics/aspect/blob/0b4a076f91ac6a9679abe126cbd0116a055fa2ee/.github/workflows/docker.yml#L10. 

The overwriting PR in the case above was https://github.com/geodynamics/aspect/actions/runs/3129699293.
I mostly make this issue to remind us of the process in case it comes up again in the future.

@tjhei: Makes sense to you? Or should we look for an alternate behavior?

**Comment by tjhei** [comment]
This makes sense and is the correct behavior, I agree. I was referring to a different issue though: 
docker build github action currently fails occasionally with "egress is over the account limit". Seems to be a github bug: https://github.com/community/community/discussions/8535

**Comment by bangerth** [comment]
I put @gassmoeller 's description above into a comment in the .yaml file (see #5203). Let's close this PR over it, and if we ever need to address the issue @tjhei had, we'll reopen it (or open another issue).
